### ISSUE TITLE
Spyder "Kernel died, restarting" issue when we run "linear_model" library

### ISSUE THREAD

**Issue by arvi79** [spyder-ide/spyder#7780]
### Issue Report Checklist

* [Y] Searched the [issues page](https://github.com/spyder-ide/spyder/issues?q=is%3Aissue) for similar reports
* [Y ] Read the relevant sections of the [Spyder Troubleshooting Guide](https://github.com/spyder-ide/spyder/wiki/Troubleshooting-Guide-and-FAQ) and followed its advice
* [Y ] Reproduced the issue after updating with ``conda update spyder`` (or ``pip``, if not using Anaconda)
* [ ] Could not reproduce inside ``jupyter qtconsole`` (if console-related)
* [ Y] Tried basic troubleshooting (if a bug/error)
    * [Y ] Restarted Spyder
    * [ ] Reset preferences with ``spyder --reset``
    * [Y ] Reinstalled the latest version of [Anaconda](https://www.anaconda.com/download/)
    * [ Y] Tried the other applicable steps from the Troubleshooting Guide
* [ ] Completed the **Problem Description**, **Steps to Reproduce** and **Version** sections below

## Problem Description
"Kernel Died, restarting" error when I run the code with linear_model in scikit learn

### What steps reproduce the problem?

1.
# Fitting Simple Linear Reg to the Training Set
from sklearn.linear_model import LinearRegression
regressor = LinearRegression()
regressor.fit(X_train, Y_train)

### What is the expected output? What do you see instead?
Expected output should be linear_regression output. I see the following 

Kernel died, restarting

 
Kernel died, restarting

### Paste Traceback/Error Below (if applicable)

```python-traceback

Spyder Internal Console

This console is used to report application
internal errors and to inspect Spyder
internals with the following commands:
  spy.app, spy.window, dir(spy)

Please don't use it to run your code

>>> WARNING:traitlets:kernel died: 18.001025199890137
WARNING:traitlets:kernel restarted


```

## Versions

* Spyder version: 3.3.1
* Python version: 3.6
* Qt version: 5.9.4
* PyQt version: 5.9.2
* Operating System name/version: Windows 7 Ultimate

### Dependencies

```
IPython >=4.0     :  6.4.0 (OK)
cython >=0.21     :  0.28.2 (OK)
jedi >=0.9.0      :  0.12.0 (OK)
matplotlib >=2.0.0:  2.2.2 (OK)
nbconvert >=4.0   :  5.3.1 (OK)
numpy >=1.7       :  1.14.3 (OK)
pandas >=0.13.1   :  0.23.0 (OK)
pycodestyle >=2.3 :  2.4.0 (OK)
pyflakes >=0.6.0  :  1.6.0 (OK)
pygments >=2.0    :  2.2.0 (OK)
pylint >=0.25     :  1.8.4 (OK)
qtconsole >=4.2.0 :  4.3.1 (OK)
rope >=0.9.4      :  0.10.7 (OK)
sphinx >=0.6.6    :  1.7.4 (OK)
sympy >=0.7.3     :  1.1.1 (OK)

```

**Comment by ccordoba12** [comment]
Since X_train and Y_train are not available in the code you posted, we can't reproduce your error. My guess is that those sets are too big, leading to a very high RAM consumption, which makes the kernel crash and restart.

**Comment by arvi79** [comment]
My dataset is a very small one (attaching it herewith)
[Salary_Data XLSX.xlsx](https://github.com/spyder-ide/spyder/files/2334339/Salary_Data.XLSX.xlsx)


Here is my entire code -


import numpy as np
import matplotlib.pyplot as plt
import pandas as pd

#importing the dataset
dataset = pd.read_csv("Salary_Data.csv")
X = dataset.iloc[:,:-1].values
Y = dataset.iloc[:,1].values


# Splitting the data into Training and Test set
from sklearn.model_selection import train_test_split
X_train, X_test, Y_train, Y_test = train_test_split (X,Y, test_size = 1/3, random_state = 0)

# Fitting Simple Linear Reg to the Training Set
from sklearn.linear_model import LinearRegression
regressor = LinearRegression()
regressor.fit(X_train, Y_train)

**Comment by ccordoba12** [comment]
Ok, thanks. We'll take a look at it in our next release.

**Comment by bcolsen** [comment]
I cannot reproduce this on latest Anconda:

    Spyder version: 3.3.1
    Python version: 3.6.6
    Qt version: 5.9.6
    PyQt version: 5.9.2
    Operating System name/version: Windows 10(virtual)
    Scikit-Learn version: 0.19.1


```
IPython >=4.0     :  6.5.0 (OK)
cython >=0.21     :  0.28.5 (OK)
jedi >=0.9.0      :  0.12.1 (OK)
matplotlib >=2.0.0:  2.2.3 (OK)
nbconvert >=4.0   :  5.3.1 (OK)
numpy >=1.7       :  1.15.1 (OK)
pandas >=0.13.1   :  0.23.4 (OK)
pycodestyle >=2.3 :  2.4.0 (OK)
pyflakes >=0.6.0  :  2.0.0 (OK)
pygments >=2.0    :  2.2.0 (OK)
pylint >=0.25     :  2.1.1 (OK)
qtconsole >=4.2.0 :  4.4.1 (OK)
rope >=0.9.4      :  0.11.0 (OK)
sphinx >=0.6.6    :  1.7.7 (OK)
sympy >=0.7.3     :  1.2 (OK)

```

**Comment by ccordoba12** [comment]
@arvi79, are you using Anaconda?

**Comment by ccordoba12** [comment]
@arvi79, does your code work in the Jupyter notebook?

**Comment by arvi79** [comment]
Same issue in Jupyter Notebook also.

**Comment by ccordoba12** [comment]
Then this is not a problem with Spyder but with Pandas or Scikit-learn. So I'm closing this one, sorry.
